app/screens/receive-bitcoin-screen/receive-bitcoin-screen.tsx

```diff
@@ -350,18 +350,17 @@ export const ReceiveBitcoinScreen = ({ navigation }: Props) => {
     const getFullUri = (input) => {
       if (type === "lightning") {
         // TODO add lightning:
-        return input
+        return input.toUpperCase()
       }
       const uri = `bitcoin:${input}`
       const params = new URLSearchParams()
-      if (amount) {
-        params.append("amount", `${amount / 10 ** 8}`)
-      }
+      if (amount) params.append("amount", `${amount / 10 ** 8}`)
       if (memo) {
         params.append("message", encodeURI(memo))
+        return `${uri}?${params.toString()}`
```

Comment:
I think this is what BTCPayServer is doing though? the mix encoding? in case they do it I would suggest following this for now.

Reply:
ie: not capitalizing url. (not sure why they're doing it now. but we should probably follow their guideline as of now)

Reply:
Ok so to confirm I should have everything uppercase except the message value? And it will automatically create the most efficient encoding.

`BITCOIN:BC1Q0....D54J0?AMOUNT=0.10779337&MESSAGE=Hey`

@jrbuhl93 @nicolasburtey

Reply:
That's what I was thinking, but amount and message should be lower case. [This](https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses) says, "Unfortunately, the ? and & needed for passing additional parameters in a BIP21 URI are not part of the QR code uppercase character set, so only binary mode can be used for those characters. Additionally, BIP21 specifies that query parameter names such as amount and label are case sensitive, so uppercase versions of them aren’t expected to work anyway."

Reply:
ok thanks @jrbuhl93, makes sense. 

I think we are there now.

- Lightning invoice data is uppercase (but not when copied from clipboard)
- Bitcoin invoice data is uppercase up until the `?` to enable mixed encoding (but not when copied from clipboard)